server/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```
@@ -129,19 +147,28 @@ public RegexpQueryBuilder flags(RegexpFlag... flags) {
                 value |= flag.value;
             }
         }
-        this.flagsValue = value;
+        this.syntaxFlagsValue = value;
         return this;
     }
 
     public RegexpQueryBuilder flags(int flags) {
-        this.flagsValue = flags;
+        this.syntaxFlagsValue = flags;
         return this;
     }
 
     public int flags() {
-        return this.flagsValue;
+        return this.syntaxFlagsValue;
     }
+    
+    public RegexpQueryBuilder caseInsensitive(boolean caseInsensitive) {
+        this.caseInsensitive = caseInsensitive;
```

We should throw an error if the value is set to `false` ?